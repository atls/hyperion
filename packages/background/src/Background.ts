import { backgroundImage, backgroundSize, system } from 'styled-system'

import styled                                      from '@emotion/styled'
import { Box }                                     from '@atlantis-lab/layout'

export const Background = styled(Box)<any>(
  ({ backgroundColor }) => ({
    position: 'fixed',
    width: '100%',
    height: '100%',
    backgroundColor,
  }),
  backgroundSize,
  backgroundImage,
  system({
    cursor: true,
  })
)

Background.defaultProps = {
  display: 'flex',
  backgroundPosition: 'center',
  backgroundRepeat: 'no-repeat',
  backgroundSize: 'cover',
}
