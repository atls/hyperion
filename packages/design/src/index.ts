export * from './Sample'
export * from './Layout'
export * from './Container'
export * from './List'
export * from './Text'
export * from './Space'
