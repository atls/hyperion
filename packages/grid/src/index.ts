export * from './Column'
export * from './Grid'
