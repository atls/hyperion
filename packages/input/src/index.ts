export * from './Textarea'
export * from './Input'
export * from './Checkbox'
