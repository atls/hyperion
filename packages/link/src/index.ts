export * from './Link'
export * from './NextLink'
export * from './RouteLink'
