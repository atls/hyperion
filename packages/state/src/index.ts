export * from './withState'
export * from './useActive'
export * from './useHover'
export * from './useFocus'
