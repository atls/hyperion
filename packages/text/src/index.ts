export * from './Text'
export * from './Space'
export * from './SearchResult'
