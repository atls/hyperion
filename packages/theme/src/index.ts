import * as theme from './theme'

export * from './Provider'
export * from './globals'

export { theme }
