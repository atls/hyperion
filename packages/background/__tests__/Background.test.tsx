import React          from 'react'
import renderer       from 'react-test-renderer'

import { Background } from '../src/index'

it('Background should render correctly', () => {
  const tree = renderer.create(<Background backgroundColor='#00f' />).toJSON()

  expect(tree).toMatchSnapshot()
})
