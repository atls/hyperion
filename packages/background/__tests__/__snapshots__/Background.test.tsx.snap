// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Background should render correctly 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 100%;
  background-color: #00f;
  position: fixed;
  background-color: #00f;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: center;
}

<div
  className="emotion-0"
  display="flex"
  height="100%"
  width="100%"
/>
`;
